nixos/modules/services/web-servers/unit/default.nix

```diff
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.unit;
+
+  configFile = builtins.toFile "unit.json" ''
+    ${cfg.config}
+  '';
+
+in {
+  options = {
+    services.unit = {
+      enable = mkEnableOption "Unit App Server";
+      package = mkOption {
+        type = types.package;
+        default = pkgs.unit;
+        description = "Unit package to use.";
+      };
+      stateDir = mkOption {
+        default = "/var/spool/unit";
+        description = "Unit data directory.";
+      };
+      logDir = mkOption {
+        default = "/var/log/unit";
+        description = "Unit log directory.";
+      };
+      config = mkOption {
+        type = types.lines;
+        default = ''
+          {
+            "listeners": {},
+            "applications": {}
+          }
+        '';
+        description = "Unit configuration in JSON format.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.tmpfiles.rules = [
+      "d ${cfg.stateDir} 0755 root root - -"
+      "d ${cfg.logDir} 0755 root root - -"
+      "d /run/unit 0755 root root - -"
```

Comment:
In the case of a hard code path like /run/unit instead of tmpfiles.d you should probably use:
serviceConfig.RuntimeDirectory = "unit";

Reply:
Thanks, update.